lib/class-wp-rest-autosaves-controller.php

```diff
@@ -339,6 +349,18 @@ public function prepare_item_for_response( $post, $request ) {
 
 		$response = $this->revisions_controller->prepare_item_for_response( $post, $request );
 
+		$schema = $this->get_item_schema();
+
+		if ( ! empty( $schema['properties']['preview_link'] ) ) {
+			$response->data['preview_link'] = get_preview_post_link( $post->post_parent, array(
+				'preview_id'    => $post->post_parent,
```

Comment:
Is `preview_id` supposed to be `$post->post_parent`? This ends up as `0` on a draft post:

<img width="997" alt="image" src="https://user-images.githubusercontent.com/36432/41064309-0c52699a-6990-11e8-8d32-5d5a5eeccd96.png">

Reply:
>Is `preview_id` supposed to be `$post->post_parent`? This ends up as `0` on a draft post:

Good call. We might consider using something like [`wp_is_post_autosave`](https://developer.wordpress.org/reference/functions/wp_is_post_autosave/) to vary the logic. Though in this context since we assume the `$post` is the result of either an autosave (`get_item`) or generated by an autosave intent, maybe just testing its type `$post->post_type === 'revision'`.

Reply:
Hmm the `-autosave` post name suffix checking in `wp_is_post_autosave` makes me think it might be better to rely on that. Guessing it might be possible for a non-autosave revision to be attached to a post parent?